src/PrestaShopBundle/Routing/Linter/NamingConventionLinter.php

```diff
@@ -66,8 +65,10 @@ public function lint($routeName, Route $route)
         ]);
 
         if ($routeName !== $expectedRouteName) {
-            throw new NamingConventionException(sprintf('Route "%s" does not follow naming convention.', $routeName));
+            return $expectedRouteName;
```

Comment:
I prefer keeping the exception, but you could add a new parameter in this exception to get back the expected route name

Reply:
Sorry by `I prefer` (bad use of words) I meant we want to avoid having methods with multi type returns, the scope of the function is then not instantly clear and requires to check the type returned by the function to check if it succeeded or not
Rather a smart use of exception is much clearer which is why I suggested to stick to it, but add the required information in it Then it can be caught and used for the appropriate purpose :)